File: week6/As2.m
x = [];
y = [];
xl = input('Enter length of x: ');
yl = input('Enter length of y: ');
disp('Enter elements into x:');
for i = 1:xl
    a = input('Enter: ');
    x = [x a];
end
disp('Enter elements into y:');
for i = 1:yl
    a = input('Enter: ');
    y = [y a];
end
z = [];
for k = 0 : yl - 2
    s = 0;
    for n = 1 : xl
        if n + k <= xl
            s = s + x(n) * y(n + k);
        end
    end
    z = [z s];
end
plot(z);
xlabel('Lag');
ylabel('Cross-correlation');
title('Cross-correlation between x and y');

File: week6/As2.py
import matplotlib.pyplot as plt
import numpy as np
x = []
y = []
xl = int(input("Enter length of x: "))
yl = int(input("Enter length of y: "))
print("Enter elements into x:")
for i in range(xl):
    a = int(input("Enter: "))
    x.append(a)   
print("Enter elements into y:")
for i in range(yl):
    a = int(input("Enter: "))
    y.append(a)
z = []
for k in range(yl - 1):
    s = 0
    for n in range(xl):
        if n + k < xl:
            s += x[n] * y[n + k]
    z.append(s)
plt.plot(z)
plt.xlabel('Lag')
plt.ylabel('Cross-correlation')
plt.title('Cross-correlation between x and y')
plt.show()
